### PR TITLE
[Compute] disk create: add --disk-iops-read-only, --disk-mbps-read-only, --max-shares, --image-reference, --image-reference-lun, --gallery-image-reference, --gallery-image-reference-lun

### PR DESCRIPTION
**Pull request by qwordy**
#12380 

Changes:
--disk-iops-read-only
The total number of IOPS that will be allowed across all VMs mounting the shared disk as ReadOnly. One operation can transfer between 4k and 256k bytes

disk_mbps_read_only
The total throughput (MBps) that will be allowed across all VMs mounting the shared disk as ReadOnly. MBps means millions of bytes per second - MB here uses the ISO notation, of powers of 10

--image-reference
ID or URN (publisher:offer:sku:version) of the image from which to create a disk

--image-reference-lun
If the disk is created from an image's data disk, this is an index that indicates which of the data disks in the image to use. For OS disks, this field is null

--gallery-image-reference
ID of the shared galley image version from which to create a disk

--gallery-image-reference-lun
If the disk is created from an image's data disk, this is an index that indicates which of the data disks in the image to use. For OS disks, this field is null

--max-shares
The maximum number of VMs that can attach to the disk at the same time. Value greater than one indicates a disk that can be mounted on multiple VMs at the same time

Examples:
```
az disk create -g rg -n disk --size-gb 10 --sku UltraSSD_LRS --disk-iops-read-only 200 --disk-mbps-read-only 30
az disk create -g rg -n disk --image-reference /Subscriptions/<subs_id>/Providers/Microsoft.Compute/Locations/westus/Publishers/Canonical/ArtifactTypes/VMImage/Offers/UbuntuServer/Skus/18.04-LTS/Versions/18.04.202002180
az disk create -g rg -n disk --image-reference Canonical:UbuntuServer:18.04-LTS:18.04.202002180
az disk create -g rg -n disk --gallery-image-reference /subscriptions/<subs_id>/resourceGroups/<rg_id>/providers/Microsoft.Compute/galleries/gallery/images/image/versions/1.0.0
az disk create -g rg -n disk --size-gb 256 --max-shares 2 -l centraluseuap
```

**History Notes:**  
(Fill in the following template if multiple notes are needed, otherwise PR title will be used for history note.)

[Compute] upgrade disk API version to 2019-11-01
[Compute] disk create: add --disk-iops-read-only, --disk-mbps-read-only, --max-shares, --image-reference, --image-reference-lun, --gallery-image-reference, --gallery-image-reference-lun

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by haroldrandom** [comment]
No YAML re-recording?

**Comment by qwordy** [comment]
> 
> 
> No YAML re-recording?

in progress

File: src/azure-cli-core/azure/cli/core/profiles/_shared.py [hunk 1]
```diff
@@ -127,7 +127,7 @@ def default_api_version(self):
         ResourceType.MGMT_NETWORK: '2019-11-01',
         ResourceType.MGMT_COMPUTE: SDKProfile('2019-07-01', {
             'resource_skus': '2019-04-01',
-            'disks': '2019-07-01',
+            'disks': '2019-11-01',
```
Comment (yungezz) [inline src/azure-cli-core/azure/cli/core/profiles/_shared.py]: have we done full regression :)

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 2]
```diff
@@ -134,6 +134,13 @@ def load_arguments(self, _):
         c.argument('disk_mbps_read_write', type=int, min_api='2018-06-01', help="The bandwidth allowed for this disk. Only settable for UltraSSD disks. MBps means millions of bytes per second with ISO notation of powers of 10")
         c.argument('upload_size_bytes', type=int, min_api='2019-03-01',
                    help='The size (in bytes) of the contents of the upload including the VHD footer. Min value: 20972032. Max value: 35183298347520')
+        c.argument('max_shares', type=int, help='The maximum number of VMs that can attach to the disk at the same time. Value greater than one indicates a disk that can be mounted on multiple VMs at the same time')
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py]: is there constraint on this int value?

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 3]
```diff
@@ -134,6 +134,13 @@ def load_arguments(self, _):
...
+        c.argument('disk_iops_read_only', type=int, help='The total number of IOPS that will be allowed across all VMs mounting the shared disk as ReadOnly. One operation can transfer between 4k and 256k bytes')
+        c.argument('disk_mbps_read_only', type=int, help='The total throughput (MBps) that will be allowed across all VMs mounting the shared disk as ReadOnly. MBps means millions of bytes per second - MB here uses the ISO notation, of powers of 10')
+        c.argument('image_reference', help='ID of the image from which to create a disk')
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py]: support name?

File: src/azure-cli/azure/cli/command_modules/vm/_params.py (hunk 2 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py]: Hard to put it into help. Too long.
#name: "P15", minsizeGiB: 128, maxsizeGiB: 256, miniops: 1100, maxiops: 1100, minbandwidthMBps: 125, maxbandwidthMBps: 125, maxValueOfMaxShares: 2
#name: "P20", minsizeGiB: 256, maxsizeGiB: 512, miniops: 2300, maxiops: 2300, minbandwidthMBps: 150, maxbandwidthMBps: 150, maxValueOfMaxShares: 2
#name: "P30", minsizeGiB: 512, maxsizeGiB: 1024, miniops: 5000, maxiops: 5000, minbandwidthMBps: 200, maxbandwidthMBps: 200, maxValueOfMaxShares: 5
#name: "P40", minsizeGiB: 1024, maxsizeGiB: 2048, miniops: 7500, maxiops: 7500, minbandwidthMBps: 250, maxbandwidthMBps: 250, maxValueOfMaxShares: 5
#name: "P50", minsizeGiB: 2048, maxsizeGiB: 4096, miniops: 7500, maxiops: 7500, minbandwidthMBps: 250, maxbandwidthMBps: 250, maxValueOfMaxShares: 5
#name: "P60", minsizeGiB: 4096, maxsizeGiB: 8192, miniops: 16000, maxiops: 16000, minbandwidthMBps: 500, maxbandwidthMBps: 500, maxValueOfMaxShares: 10
#name: "P70", minsizeGiB: 8192, maxsizeGiB: 16384, miniops: 18000, maxiops: 18000, minbandwidthMBps: 750, maxbandwidthMBps: 750, maxValueOfMaxShares: 10
#name: "P80", minsizeGiB: 16384, maxsizeGiB: 32767, miniops: 20000, maxiops: 20000, minbandwidthMBps: 900, maxbandwidthMBps: 900, maxValueOfMaxShares: 10

File: src/azure-cli-core/azure/cli/core/profiles/_shared.py (hunk 1 above)
Comment (qwordy) [reply src/azure-cli-core/azure/cli/core/profiles/_shared.py]: underway

File: src/azure-cli/azure/cli/command_modules/vm/_params.py (hunk 3 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py]: OK. Plan to support ID, name, URN

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py]: Support ID and URN now. Name (custom image) is not support by service.

File: src/azure-cli-core/azure/cli/core/profiles/_shared.py (hunk 1 above)
Comment (qwordy) [reply src/azure-cli-core/azure/cli/core/profiles/_shared.py]: Done. Can't guarantee no error in live test. But no new error in disk related tests.

**Pull request merged**